src/Chapter1.hs

```diff
@@ -542,7 +542,14 @@ value after "=" where the condition is true.
 Casual reminder about adding top-level type signatures for all functions :)
 -}
 
-mid x y z = error "mid: not implemented!"
+mid :: Int -> Int -> Int -> Int
+mid x y z
+    | x <= y && y <= z = y
+    | x <= z && z <= y = z
+    | y <= x && x <= z = x
+    | y <= z && z <= x = z
+    | z <= x && x <= y = x
+    | otherwise        = y
```

Comment:
As we mentioned, the compiler in Haskell is very attentive to the exhaustive pattern-matching. And here it would warn you that `Pattern matching is not exhaustive` if you don't put the `otherwise` case, as the guards have quite complicated logic, and the compiler won't be able to prove that it covers all the cases.

However, you can simplify your cases a bit. The first case `x <= y && y <= z = y` could be covered by this `otherwise`. So you can safely remove it 👌🏼 
The implementation will be a bit shorter 🙂